packages/host/app/components/operator-mode/code-mode.gts

```diff
@@ -49,41 +45,73 @@ export default class CodeMode extends Component<Signature> {
     return this.realmInfo?.iconURL;
   }
 
+  get codePath() {
+    return this.operatorModeStateService.state.codePath;
+  }
+
   @action resetLoadFileError() {
     this.loadFileError = null;
   }
 
-  fetchCodeModeRealmInfo = restartableTask(async () => {
-    if (!this.operatorModeStateService.state.codePath) {
-      return;
-    }
+  @action loadRealmInfo() {
+    this.realmInfoResource.load();
+  }
+
+  @use realmInfoResource = resource(() => {
```

Comment:
This will reload whenever `codePath` changes

Reply:
The reason I used the `task` instead of `resource` here is to prevent the realm info from being reloaded if the new `codePath` is still within the same realm.

With resource:


https://github.com/cardstack/boxel/assets/12637010/d4492556-3437-4e64-bcc7-37d757e2ca4d


With task:


https://github.com/cardstack/boxel/assets/12637010/58e9c119-26f2-42c4-933c-813bf74abbc2



When using `resource`, the background and realm info will always be reloaded, regardless of whether the URL's realm changes or not. If we are okay with the background and realm info in the card bar URL being always reloaded then `resource` is fine.

Reply:
Nice catch. I'll see if I can adjust the resource to not reload the info when the realm is the same

Reply:
Addressed this in https://github.com/cardstack/boxel/pull/599/commits/1e03f239f76dfe3a43ddeb9a0e1c394b91961da8